src/connectors/jamf_inventory.py

```diff
@@ -75,7 +82,39 @@ async def main(table_name):
 
 def ingest(table_name, options):
     global HEADERS
-    creds = options.get('credentials', '')
-    HEADERS = {'Authorization': f'Basic {creds}', 'Accept': 'application/json'}
+    token = getAccessToken(options=options['credentials'])
+    HEADERS = {'Authorization': f'Bearer {token}', 'Accept': 'application/json'}
     return asyncio.get_event_loop().run_until_complete(main(f'data.{table_name}'))
 
+
+# options is a dict containing cliendId and clientSecret
+def getAccessToken(credentials: str) -> str:
+
+    # credentials stored in format {"client_id":"some-id","client_secret":"some-secret"}
+    client_cred = json.loads(credentials)
+
+    jamf_token_api_path = "https://snowflake.jamfcloud.com/api/oauth/token"
+
+    data = {
+        'client_id': client_cred[CLIENT_ID_KEY],
+        'grant_type': 'client_credentials',
+        'client_secret': client_cred[CLIENT_SECRET_KEY],
+    }
+
+    response = requests.post(
+        jamf_token_api_path,
+        data=urlencode(data),
+        headers={"Content-Type": "application/x-www-form-urlencoded"},
+    )
+
+    json_response = response.json()
+
+    if ACCESS_TOKEN_KEY in json_response:
+        return json_response[ACCESS_TOKEN_KEY]
+
+    log.error(
+        f"{ACCESS_TOKEN_KEY} not found in response from api : {jamf_token_api_path}"
```

Comment:
I'm not sure what `log.error` does in our case but it seems likely it's better to throw an exception so that it can be tracked in an error tracker like Sentry